shared/util.h

```diff
@@ -92,7 +92,21 @@ static inline void freep(void *p)
 }
 #define _cleanup_free_ _cleanup_(freep)
 
-static inline bool addu64_overflow(uint64_t a, uint64_t b, uint64_t *res)
+static inline bool add_u32_overflow(uint32_t a, uint32_t b, uint32_t *res)
+{
+#if HAVE___BUILTIN_UADD_OVERFLOW
+#if __SIZEOF_INT__ == 4
+	return __builtin_uadd_overflow(a, b, res);
+#else
+#error "sizeof(int) != 4"
+#endif
```

Comment:
You could just remove the else-error-block here. It would fall through to the current code.